mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapperTest.java

```diff
@@ -0,0 +1,102 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.modules.CassandraAnnotationModule;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+public class CassandraAnnotationMapperTest {
+    private static final MailboxAnnotation SHARED_ANNOTATION = new MailboxAnnotation("/shared/comment", "My shared comment");
+    private static final MailboxAnnotation PRIVATE_ANNOTATION_UPDATE = new MailboxAnnotation("/private/comment", "My updated private comment");
+    private static final MailboxAnnotation PRIVATE_ANNOTATION = new MailboxAnnotation("/private/comment", "My private comment");
+    private CassandraAnnotationMapper cassandraAnnotationMapper;
+    private CassandraCluster cassandra;
+    private CassandraId mailboxId;
+
+    @Before
+    public void setUp() throws Exception {
+        cassandra = CassandraCluster.create(new CassandraAnnotationModule());
+        cassandra.ensureAllTables();
+        mailboxId = CassandraId.of(UUID.fromString("464765a0-e4e7-11e4-aba4-710c1de3782b"));
+        cassandraAnnotationMapper = new CassandraAnnotationMapper(mailboxId, cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cassandra.clearAllTables();
+    }
+
+    @Test
+    public void retrieveEmptyAnnotation() throws Exception {
+        assertThat(cassandraAnnotationMapper.getAnnotations()).isEmpty();
+    }
+
+    @Test
+    public void retrieveAnnotation() throws Exception {
+        cassandraAnnotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
+        List<MailboxAnnotation> expectedResult = Lists.newArrayList(PRIVATE_ANNOTATION);
+        assertThat(cassandraAnnotationMapper.getAnnotations()).isEqualTo(expectedResult);
+    }
+
+    @Test
+    public void retrieveMultiAnnotations() throws Exception {
+        cassandraAnnotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
+        cassandraAnnotationMapper.insertAnnotation(SHARED_ANNOTATION);
+
+        assertThat(cassandraAnnotationMapper.getAnnotations()).isEqualTo(Lists.newArrayList(PRIVATE_ANNOTATION, SHARED_ANNOTATION));
+    }
+
+    @Test
+    public void addAnnotationWhenStoredShouldReturnUpdatedAnnotation() throws Exception {
+        cassandraAnnotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
+
+        assertThat(cassandraAnnotationMapper.getAnnotations()).isEqualTo(Lists.newArrayList(PRIVATE_ANNOTATION));
```

Comment:
Any test should have at most 3 sections : Given, When and Then. Here, you have to Then section.